src/StrokeEngine.cpp

```diff
@@ -557,169 +183,43 @@ String StrokeEngine::getPatternName(int index) {
-
-    } else {
-        // Move MAX_TRAVEL towards the homing switch
-        servo->move(-_motor->stepsPerMillimeter * _physics->physicalTravel * _homeingToBack);
-    }
-
-    // Poll homing switch
-    while (servo->isRunning()) {
-
-        // Switch is active low
-        if (digitalRead(_homeingPin) == !_homeingActiveLow) {
-
-            // Set home position
-            if (_homeingToBack == 1) {
-                //Switch is at -KEEPOUT_BOUNDARY
-                servo->forceStopAndNewPosition(-_motor->stepsPerMillimeter * _physics->keepoutBoundary);
-
-                // drive free of switch and set axis to lower end
-                servo->moveTo(_minStep);
-
-            } else {
-                servo->forceStopAndNewPosition(_motor->stepsPerMillimeter * (_physics->physicalTravel - _physics->keepoutBoundary));
-
-                // drive free of switch and set axis to front end
-                servo->moveTo(_maxStep);
-            }
-            _isHomed = true;
-
-            // drive free of switch and set axis to 0
-            servo->moveTo(0);
-            
-            // Break loop, home was found
-            break;
-        }
-
-        // Pause the task for 20ms to allow other tasks
-        vTaskDelay(20 / portTICK_PERIOD_MS);
-    }
-    
-    // disable Servo if homing has not found the homing switch
-    if (!_isHomed) {
-        servo->disableOutputs();
-        _state = UNDEFINED;
-
-#ifdef DEBUG_TALKATIVE
-        Serial.println("Homing failed");
-#endif
-
-    } else {
-        // Set state to ready
-        _state = READY;
-
-#ifdef DEBUG_TALKATIVE
-        Serial.println("Homing succeeded");
-#endif
-    }
-
-    // Call notification callback, if it was defined.
-    if (_callBackHomeing != NULL) {
-        _callBackHomeing(_isHomed);
-    }
-
-    // Set first point for telemetry
-    if (_callbackTelemetry != NULL) {
-        _callbackTelemetry(0.0, 0.0, false);
-    }
-
-#ifdef DEBUG_TALKATIVE
-    Serial.println("Stroke Engine State: " + verboseState[_state]);
-#endif
-
-    // delete one-time task
-    _taskHomingHandle = NULL;
-    vTaskDelete(NULL);
-}
-
 void StrokeEngine::_stroking() {
     motionParameter currentMotion;
 
     while(1) { // infinite loop
 
-        // Suspend task, if not in PATTERN state
-        if (_state != PATTERN) {
-            vTaskSuspend(_taskStrokingHandle);
+        // Suspend task, if motor is not active
+        // TODO - This doesn't look quite right
+        if (!this->motor->isInState(MotorState::ACTIVE)) {
+          ESP_LOGI("StrokeEngine", "Motor is no longer active! Attempting to suspend pattern.");
+          this->stopPattern();
         }
 
         // Take mutex to ensure no interference / race condition with communication threat on other core
         if (xSemaphoreTake(_patternMutex, 0) == pdTRUE) {
 
-            if (_applyUpdate == true) {
+            if (this->applyUpdate == true) {

```

Comment:
Is there a reason you use this-> ?

Reply:
Sorry for the noob question

Reply:
It's just habit coming from NodeJS `this.`, and having built previous c++ code using `this->`. I tend to prefer being explicit when a object member is being referenced, as it makes it easier to read on my eyes.

I'm not sure what the common c++ community stylistic recommendations are with this, but I'll look into it more when I add a linter.